### ISSUE TITLE
EntityClientEndpoint stuck on close after receiving exception response to invoke

### ISSUE THREAD

**Issue by chrisdennis** [Terracotta-OSS/terracotta-core#149]
In one of the Ehcache tests I get a blocked endpoint close after the previous entity invoke failed with an exception:
```
Stack Trace
Test worker [12] (WAITING)
   java.lang.Object.wait line: not available [native method]
   java.lang.Object.wait line: 502 
   com.tc.object.InFlightMessage.waitForAcks line: 88 
   com.tc.object.ClientEntityManagerImpl.createInFlightMessageAfterAcks line: 408 
   com.tc.object.ClientEntityManagerImpl.synchronousWaitForResponse line: 381 
   com.tc.object.ClientEntityManagerImpl.internalRelease line: 366 
   com.tc.object.ClientEntityManagerImpl.access$300 line: 70 
   com.tc.object.ClientEntityManagerImpl$2.run line: 326 
   com.tc.object.EntityClientEndpointImpl.close line: 197 
   org.ehcache.clustered.client.EhcacheClientEntity.close line: 49 
   org.ehcache.clustered.client.EhcacheClientEntityFactory.retrieve line: 124 
   org.ehcache.clustered.EhcacheClientEntityFactoryIntegrationTest.testRetrieveWithBadConfig line: 128 
```

**Comment by chrisdennis** [comment]
I'm still seeing this with snapshots pulled this morning.

**Comment by myronkscott** [comment]
I was unable to reproduce this bug last night. Jeff mentioned a test case merged in galvan that also exhibits this behavior. I will be taking a closer look later in the day.

**Comment by chrisdennis** [comment]
The test case in Galvan exposes the slow connection close, not the blocked endpoint close.

**Comment by myronkscott** [comment]
is the preceding exception being thrown in on the server side of the entity?  Is this crashing the server?

**Comment by chrisdennis** [comment]
You are right, I'm an idiot... we can/will discuss later.

**Comment by chrisdennis** [comment]
This is fixed in my PR for ehcache/ehcache3#724 PR.
